toolkit/ar/src/main/java/com/arcgismaps/toolkit/ar/internal/CalibrationState.kt

```diff
@@ -0,0 +1,74 @@
+/*
+ *
+ *  Copyright 2025 Esri
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package com.arcgismaps.toolkit.ar.internal
+
+import androidx.compose.runtime.Stable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableDoubleStateOf
+import androidx.compose.runtime.setValue
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.asSharedFlow
+import kotlinx.coroutines.launch
+
+
+/**
+ * State holder class for calibration view
+ *
+ * @since 200.7.0
+ */
+@Stable
+internal class CalibrationState(private val scope: CoroutineScope) {
+    private val _headingDeltas: MutableSharedFlow<Double> = MutableSharedFlow()
+    val headingDeltas = _headingDeltas.asSharedFlow()
+
+    private val _elevationDeltas: MutableSharedFlow<Double> = MutableSharedFlow()
+    val elevationDeltas = _elevationDeltas.asSharedFlow()
+
+    var totalHeadingOffset by mutableDoubleStateOf(0.0)
+    var totalElevationOffset by mutableDoubleStateOf(0.0)
+
+    fun onHeadingChange(value: Double) {
+        scope.launch {
+            _headingDeltas.emit(value)
+            totalHeadingOffset += value
+        }
+    }
+
+    fun onElevationChange(value: Double) {
+        scope.launch {
+            _elevationDeltas.emit(value)
+            totalElevationOffset += value
+        }
+    }
+
+    fun onHeadingReset() {
+        scope.launch {
+            _headingDeltas.emit(-totalHeadingOffset)
+            totalHeadingOffset = 0.0
+        }
```

Comment:
You don't need a scope if you use `tryEmit`:
```suggestion
        _headingDeltas.tryEmit(-totalHeadingOffset)
        totalHeadingOffset = 0.0
```